File: intellilab/src/app/shared/util/SearchUtil.ts
export class SearchUtils {

  // encode all white space and #
  static ReplaceSearchTerms(searchTerm: string) {
    let st: string = searchTerm;
    if ( !st) st = '';
    st = st.replace(/^\s+/,"");
    st = st.replace(/\//g, 'slash');
    st = st.replace(/\s/g, 'whitespace');
    st = st.replace(/#/g, 'pong');
    st = st.replace(/%/g, 'percent');
    st = st.replace(/\?/g, 'questionmark');
    st = st.replace(/\./g, 'period');
    return st;
  }
}

File: intellilab/somruinternal/src/main/java/com/internal/service/somruinternal/utils/SearchUtil.java
package com.internal.service.somruinternal.utils;

public class SearchUtil {

    public static String replaceIllegalSearchCharacters( String searchTerm ) {
        String st = searchTerm == null ? "" : searchTerm;
        st = st.replaceAll("slash", "/");
        st = st.replaceAll("whitespace", " ");
        st = st.replaceAll("pong", "#");
        st = st.replaceAll("percent", "%");
        st = st.replaceAll("questionmark", "?");
        st = st.replaceAll("period", ".");
        return st;
    }    
}
